master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -624,12 +628,49 @@ private[celeborn] class Master(
     context.reply(OneWayMessageResponse)
   }
 
+  private def computeUserResourceConsumption(userIdentifier: UserIdentifier)
+      : ResourceConsumption = {
+    val current = System.currentTimeMillis()
+    if (userResourceConsumptions.containsKey(userIdentifier)) {
+      val resourceConsumptionAndUpdateTime = userResourceConsumptions.get(userIdentifier)
+      if (current - resourceConsumptionAndUpdateTime._2 > metricsResourceConsumptionInterval) {
+        val newResourceConsumption = statusSystem.workers.asScala.flatMap { workerInfo =>
+          workerInfo.userResourceConsumption.asScala.get(userIdentifier)
+        }.foldRight(ResourceConsumption(0, 0, 0, 0))(_ add _)
+        userResourceConsumptions.put(userIdentifier, (newResourceConsumption, current))._1
+      } else {
+        resourceConsumptionAndUpdateTime._1
+      }
+    } else {
+      val newResourceConsumption = statusSystem.workers.asScala.flatMap { workerInfo =>
+        workerInfo.userResourceConsumption.asScala.get(userIdentifier)
+      }.foldRight(ResourceConsumption(0, 0, 0, 0))(_ add _)
+      userResourceConsumptions.put(userIdentifier, (newResourceConsumption, current))._1
+    }
+  }
+
   private def handleCheckQuota(
       userIdentifier: UserIdentifier,
       context: RpcCallContext): Unit = {
-    val userResourceConsumption = statusSystem.workers.asScala.flatMap { workerInfo =>
-      workerInfo.userResourceConsumption.asScala.get(userIdentifier)
-    }.foldRight(ResourceConsumption(0, 0, 0, 0))(_ add _)
+
+    resourceConsumptionSource.addGauge(
+      "diskFIleCount",
+      _ => computeUserResourceConsumption(userIdentifier).diskFileCount,
+      userIdentifier.toMap)
+    resourceConsumptionSource.addGauge(
+      "diskBytesWritten",
+      _ => computeUserResourceConsumption(userIdentifier).diskBytesWritten,
+      userIdentifier.toMap)
+    resourceConsumptionSource.addGauge(
+      "hdfsFileCount",
+      _ => computeUserResourceConsumption(userIdentifier).hdfsFileCount,
+      userIdentifier.toMap)
+    resourceConsumptionSource.addGauge(
+      "hdfsBytesWritten",
+      _ => computeUserResourceConsumption(userIdentifier).hdfsBytesWritten,
+      userIdentifier.toMap)
+
+    val userResourceConsumption = computeUserResourceConsumption(userIdentifier)
```

Comment:
IMHO we can move this line before those addGauge methods and reuse the object inside those methods

Reply:
> IMHO we can move this line before those addGauge methods and reuse the object inside those methods

No, here we should pass a method to addGauge,  `userResourceConsumption`  is a certain value, if pass this value, won't change anymore